### ISSUE TITLE
self updatable widget is not optimum.

### ISSUE THREAD

**Issue by ahmed2hamy** [thangmam/smoothratingbar#32]
I don't think self updatable widget is an optimum behavior in flutter, Calling setState in void functions is a second nature for a flutter developer when required and this conflicts with your package update 1.1.0

**Comment by andrsdev** [comment]
@ahmed2hamy I agree with you, that las update is not desirable, and causing a lot of logic problems. If this is not fixed soon I will implement it myself, use another package, or forever use version 1.0.4+2
